File: 1137-n-th-tribonacci-number/1137-n-th-tribonacci-number.py
class Solution:
  def __init__(self):
    self.ls = [0,1,1]
    for i in range(3, 38):
      self.ls.append(self.ls[i - 3] + self.ls[i - 2] + self.ls[i - 1])
  def tribonacci(self, n: int) -> int:
    return self.ls[n]
        

File: 1137-n-th-tribonacci-number/1137-n-th-tribonacci-number.ts
function tribonacci(n: number): number {
  if (!n) return 0;
  if (n < 3) return 1;
  let a = 0, b = 1, c = 1;
  n -= 2;
  while (n--) [a,b,c] = [b,c,a+b+c];
  return c;
};